lib/dataset/sized_dict.cpp

```diff
@@ -236,16 +236,15 @@ constexpr auto unaligned_by_dim_slice = [](const auto &coords, const auto &key,
 }
 
 template <class Key, class Value>
-std::tuple<SizedDict<Key, Value>, SizedDict<Key, Value>>
+SizedDict<Key, Value>
 SizedDict<Key, Value>::slice_coords(const Slice &params) const {
   auto coords = slice(params);
   coords.m_readonly = false;
-  SizedDict<Key, Value> attrs(coords.sizes(), {});
   for (const auto &[key, var] : *this)
     if (unaligned_by_dim_slice(*this, key, var, params))
-      attrs.set(key, coords.extract(key));
+      coords.set_aligned(key, false);
   coords.m_readonly = true;
-  return {std::move(coords), std::move(attrs)};
+  return coords;
```

Comment:
Didn't we say that `Variable::slice` might be able to handle the change of the flag when slicing? Or does it lack some information?

Reply:
Is this meaningful? When a variable isn't in a coords dict, the flag doesn't mean anything. So
```python
x = sc.arange('x', 5)
da = sc.DataArray(sc.arange('x', 3), coords={'x': x[:3]})
```
would result in an unaligned coord even though the data array was never sliced.

Reply:
It would only change the flag when removing a dimension. But there may still be edge cases where this leads to unintentional behavior?

Reply:
Right. I guess the question is whether `var[i]` is supposed to be (un)aligned in isolation.
My hunch is to stick to handling in coords.